etc/releasing.md

```diff
@@ -209,6 +208,32 @@ pushed.
   `docs/`. This may require pushing the commit to a fork of the C++ Driver
   repository and creating a pull request.
 
+## Update CHANGELOG.md for next release
+
+### For a minor release (e.g. r1.2.0):
+
+Open a PR to the master branch to add a new empty `[Unreleased]` section to the CHANGELOG.md for the next minor release. Example:
+
+```md
+## 1.3.0 [Unreleased]
+```
+
+Open a PR to the release branch (e.g. `releases/v1.2`) to add a new empty `[Unreleased]` section to the CHANGELOG.md for the next patch release. Example:
+
+```md
+## 1.2.1 [Unreleased]
+```
```

Comment:
Does this imply the main branch should never contain changelog entries for patch releases since they'd be targeting the next minor (or major) release instead?

e.g. https://github.com/mongodb/mongo-cxx-driver/pull/1107 merged 3.10.1 changelog entries to `master` then cherry-picked them to `releases/v3.10`. Should the changelog in `master` be targeting `3.11.0 [Unreleased]` instead?

Reply:
> Should the changelog in master be targeting 3.11.0 `[Unreleased]` instead?

Yes. I think the `[Unreleased]` changelog entries should track the next release on the same branch:

- Minor releases are made from the master branch. Track the next minor release in the master branch changelog.
- Patch releases are made from release branches. Track the next patch release in the release branch changelog.

That may ease updating the changelog as part of applying the changes intended for the same release.

I updated the changelog in this PR to include `## 3.11.0 [Unreleased]` and remove `## 3.10.1 [Unreleased]`.

> Does this imply the main branch should never contain changelog entries for patch releases

I think the `[Unreleased]` patch release entry should be tracked on the release branch.

However, I think it is valuable to include published patch release entries on master after the release. That adds all release entries to one file for easy searching. This follows a similar pattern as the C driver NEWS file (e.g. both 1.26.0 and 1.25.4 are included on the master branch): https://github.com/mongodb/mongo-c-driver/blob/master/NEWS.

Reply:
> I think it is valuable to include published patch release entries on master after the release.

That was unintentional/poor wording on my part. The _entries_ would still be present, but they would belong to a minor/major release section instead of a patch release section. There would be no patch release _sections_ in the main branch.

The response and changes are consistent with my understand and LGTM.

Reply:
I would prefer that whatever approach we use also minimizes the amount of editing required during the release process (ideally, just removing `[Unreleased]`). If the entry must ultimately end up in both branches, I would prefer they be written to both branches _immediately_ rather than during the release process.

With that in mind, which of the following options do you prefer?

I'm personally in favor of Option A due to less likelihood for conflicts when cherry-picking from `master` to the release branch + minimal redundancy.

## Option A: patch release section in both master and release with implicit patch entries.

On `master`:
```md
# 3.11.0 [Unreleased]

* Entry targeting 3.11.0.

# 3.10.1 [Unreleased]

* Entry targeting 3.10.1. (Implicitly included in 3.11.0.)
```
On `releases/v3.10`:
```md
# 3.10.1 [Unreleased]

* Entry targeting 3.10.1.
```

## Option B: patch release section in both master and release with explicit patch entries.

On `master`:
```md
# 3.11.0 [Unreleased]

* Entry targeting 3.11.0.
* Entry targeting 3.10.1. (Explicitly included in 3.11.0.)

# 3.10.1 [Unreleased]

* Entry targeting 3.10.1.
```
On `releases/v3.10`:
```md
# 3.10.1 [Unreleased]

* Entry targeting 3.10.1.
```

## Option C: patch release section in release only.

On `master`:
```md
# 3.11.0 [Unreleased]

* Entry targeting 3.11.0.
* Entry targeting 3.10.1. (Explicitly included in 3.11.0.)
```
On `releases/v3.10`:
```md
# 3.10.1 [Unreleased]

* Entry targeting 3.10.1.
```

Reply:
I also prefer Option A. And the detailed description is much appreciated. The release process has been updated. CHANGELOG.md has been updated to include both upcoming patch (3.10.1) and minor (3.11.0) releases.